irs_lqr/irs_lqr.py

```diff
@@ -0,0 +1,202 @@
+import numpy as np
+import time
+
+from irs_lqr.tv_lqr import TvLqr, get_solver
+
+class IrsLqrParameters:
+    """
+    Parmeters class for IrsLqr.
+
+    Q (np.array, shape n x n): cost matrix for state.
+    Qd (np.array, shape n x n): cost matrix for final state.
+    R (np.array, shape m x m): cost matrix for input.
+    x0 (np.array, shape n): initial point in state-space.
+    xd_trj (np.array, shape (T+1) x n): desired trajectory.
+    u_trj_initial (np.array, shape T x m): initial guess of the input.
+    xbound (np.array, shape 2 x n): (lb, ub) bounds on state.
+    xbound (np.array, shape 2 x m): (lb, ub) bounds on input.
+    solver (str): solver name to use for direct LQR.
+    """
+    def __init__(self):
+        self.Q = None
+        self.Qd = None
+        self.R = None
+        self.x0 = None
+        self.xd_trj = None
+        self.u_trj_initial = None
+        self.xbound = None
+        self.ubound = None
+        self.solver_name = "osqp"
+
+class IrsLqr:
+    def __init__(self, system, params):
+        """
+        Base class for iterative Randomized Smoothing LQR.
+
+        system (DynamicalSystem class): dynamics class.
+        parms (IrsLqrParameters class): parameters class.
+        """
+
+        self.system = system
+        self.params = params
+        self.check_valid_system(self.system)
+        self.check_valid_params(self.params, self.system)
+
+        self.Q = params.Q
+        self.Qd = params.Qd
+        self.R = params.R
+        self.x0 = params.x0
+        self.xd_trj = params.xd_trj
+        self.u_trj = params.u_trj_initial
+        self.xbound = params.xbound
+        self.ubound = params.ubound
+        self.solver = get_solver(params.solver_name)
+
+        self.T = self.u_trj.shape[0] # horizon of the problem
+        self.dim_x = self.system.dim_x
+        self.dim_u = self.system.dim_u
+        self.x_trj = self.rollout(self.x0, self.u_trj)
+        self.cost = self.evaluate_cost(self.x_trj, self.u_trj)
+
+        # These store iterations for plotting.
+        self.x_trj_lst = [self.x_trj]
+        self.u_trj_lst = [self.u_trj]
+        self.cost_lst = [self.cost]
+
+        self.start_time = time.time()
+
+        self.iter = 1
+
+    def check_valid_system(self, system):
+        """
+        Check if the system is valid. Otherwise, throw exception.
+        TODO(terry-suh): we can add more error checking later.        
+        """
+        if (system.dim_x == 0):
```

Comment:
nit: Python conditional statements do not need to be surrounded by brackets. 
```python
if system.dim_x == 0:

```
should be sufficient here.

Reply:
Done.